decorator.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
- [ ] The `attr_accessor` method is a convenient way to define both a getter and a setter method for an instance variable as you can easily read and modify the nameable instance variable from other parts of your program. So please use 
``attr_accessor :nameable`` after the class Decorator.

Reply:
Hello @roshan-bajgain, thank you for your review I have some worries if you can help me? If I want to use the method `correct_name ` like said in the requirements I think using the `attr_accessor :nameable` is not the right option. Or can you please give me more details about how to implement it?

Reply:
Hello @Yacoubou-seidou, using attr_accessor is the right way: You can use it below `class Decorator < Nameable` as 
`attr_accessor :nameable` so you can easily read and modify the nameable instance variable from other parts of your program. I think this helps you.

Reply:
@Yacoubou-seidou, If you don't need to access the @nameable instance variable from outside the class. you can omit the attr_accessor declaration. But I think you need to access it in future from outside the class. So it is better to use `attr_accessor :nameable`

Reply:
@roshan-bajgain for now I think it's better to follow project requirements if changes are required further I will be more then happy to make them